### PR TITLE
fix: modify getActiveKeys() to return list of previous keys

### PR DESCRIPTION
**Pull request by mattstrom**
RKT-7833

### CODE REVIEW

File: src/helpers/set-utils.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Returns the set difference of `a` minus `b`.
+ *
+ * @param a {Set|Array}
+ * @param b {Set|Array}
+ * @returns {Set}
+ */
+export function setDifference(a, b) {
```
Comment (tfoster) [inline src/helpers/set-utils.js:8]: I think the names here would be more clear if they were reversed (diffSet and copySet)? Right now they read as mutators.

File: src/store.js [hunk 2]
```diff
@@ -287,6 +307,10 @@ function EvolvStore(options) {
     activeEids = result.activeEids;
 
     const newActiveKeys = objects.flattenKeys(effectiveGenome);
+
+    previousKeys.clear();
```
Comment (tfoster) [inline src/store.js:311]: Why not store the original set before reevaluating and then diff that against the result?

File: src/store.js [hunk 3]
```diff
@@ -52,15 +53,32 @@ function getValueActive(activeKeys, key) {
...
 
-function getActiveKeys(activeKeys, prefix) {
+function getActiveKeys(activeKeys, previousKeys, prefix) {
   const result = [];
-  activeKeys.forEach(function(key) {
-    if (!prefix || strings.startsWith(key, prefix)) {
+  const previous = [];
+
+  function hasPrefix(key) {
+    return !prefix || strings.startsWith(key, prefix)
```
Comment (tfoster) [inline src/store.js:61]: Missing semicolon.

File: src/store.js [hunk 4]
```diff
@@ -52,15 +53,32 @@ function getValueActive(activeKeys, key) {
...
+    if (hasPrefix(key)) {
+      previous.push(key);
+    }
+  });
+
+  return {
+    current: result,
+    previous: previous,
+    added: setDifference(result, previous),
+    removed: setDifference(previous, result)
```
Comment (tfoster) [inline src/store.js:80]: added and removed don't seem to be used, can they just be removed?

File: src/store.js (hunk 2 above)
Comment (mattstrom) [reply src/store.js:311]: I'm not following entirely. Nevertheless, for the `getActiveKeys.bind(this, activeKeys, previousKeys))` line later on to work, `previousKeys` has to exist at the time of binding and cannot be reassigned.

File: src/store.js (hunk 4 above)
Comment (mattstrom) [reply src/store.js:80]: Yes, they can. They were used downstream but are not any longer.

**Comment by FrazerBayley** [comment]
hotfix-v1.3

**Pull request merged**